src/Sylius/Bundle/ApiBundle/Filters/SearchFilter.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Filters;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\SearchFilter as BaseSearchFilter;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use Doctrine\ORM\QueryBuilder;
+
+final class SearchFilter extends BaseSearchFilter
+{
+    public function addWhereByStrategy(
+        string $strategy,
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $alias,
+        string $field,
+        $value,
+        bool $caseSensitive
+    ) {
+        try {
+            parent::addWhereByStrategy($strategy, $queryBuilder, $queryNameGenerator, $alias, $field, $value, $caseSensitive);
+        } catch (InvalidArgumentException $invalidArgumentException) {
+            $wrapCase = $this->createWrapCase($caseSensitive);
+            $valueParameter = $queryNameGenerator->generateParameterName($field);
+
+            if ($strategy === 'partial_or' && $invalidArgumentException->getMessage() === sprintf('strategy %s does not exist.', $strategy)) {
+                $queryBuilder
+                    ->orWhere(sprintf($wrapCase('%s.%s').' LIKE '.$wrapCase('CONCAT(\'%%\', :%s, \'%%\')'), $alias, $field, $valueParameter))
+                    ->setParameter($valueParameter, $value);
+            } else {
```

Comment:
btw `else` not needed 💃 but we still should try to go with a totally custom filter